### ISSUE TITLE
Signing Region for region "local" is inconsistent with aws-sdk v2 and aws cli

### ISSUE THREAD

**Issue by sebastian-fredriksson-bernholtz** [aws/aws-sdk-js-v3#4182]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

This sdk seems to use signingRegion `us-east-1` when region is set to `local`. This is inconsistent with the behaviour of [aws-sdk v2](https://github.com/aws/aws-sdk-js) and [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) (`aws-cli/2.7.16 Python/3.10.5 Darwin/22.1.0 source/arm64 prompt/off`) and can lead to unintuitive and difficult to debug issues when working with [dynamodb-local](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html).

I think the source of this can be found [here (v.3.100.0)](https://github.com/aws/aws-sdk-js-v3/blob/v3.100.0/clients/client-dynamodb/src/endpoints.ts#L21), ([main as of time of writing - eb9ec5e](https://github.com/aws/aws-sdk-js-v3/blob/eb9ec5ec8928744d3ba484b507393cd86e3b2414/clients/client-dynamodb/src/endpoint/ruleset.ts#L356)).

It might be my lack of Google-fu and documentation skimming skills, but I really struggled to find any information about this behaviour and I spent a lot of time first debugging my issue, and then going through this codebase to understand what was going on.

If this behaviour is not a bug, but actually required for some other situation, I wonder if it could be made a bit more obvious in the documentation?

#### Background
I found this out when I started using dynamodb-toolbox - which uses aws-sdk v2 - in a project where tables are created in [dynamodb-local](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html) using `@aws-sdk/client-dynamodb` with region set to `local`. My tests failed with `ResourceNotFoundException`.

I soon found that using the `-sharedDb` setting fixed the tests. The `-sharedDb` setting determines whether dynamodb-local will use single database file for all [access key, region] combinations as opposed to a unique database file for each [access key, region] combination. With that clue, I removed both `-inMemory` and `-sharedDb` options and started experimenting and checking what database files were created. 

I noticed that a database file for `us-east-1` kept being created despite me not having set that region anywhere. At one point I thought it was being used as default when it was an unsupported region, but when I tried a completely random region, that region caused a database file to be created. That is when I started googling for `aws-sdk`, `region`, and `local`. 

In the end, after not finding much useful information I started looking through this source code, and eventually found what to me seems to be the [cause](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-dynamodb/src/endpoint/ruleset.ts#L356).

### SDK version number

@aws-sdk/client-dynamodb@3.100.0, @aws-sdk/client-dynamodb@main (eb9ec5e)

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

16.17.0

### Reproduction Steps

1. Run `docker run -p 8000:8000 amazon/dynamodb-local -jar DynamoDBLocal.jar`
2. Run following command to create Table `Test` in `1_local` DB
```bash
AWS_REGION="local" AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 aws dynamodb create-table --attribute-definitions '[{"AttributeName":"pk", "AttributeType": "S"}]' --table-name "Test" --key-schema '[{"AttributeName":"pk", "KeyType": "HASH"}]' --endpoint-url "http://localhost:8000" --no-verify-ssl --billing-mode 'PAY_PER_REQUEST'
```
3. Run following typescript code to attempt to create the same table
``` typescript
import { DynamoDB } from '@aws-sdk/client-dynamodb';

const client = new DynamoDB({
    credentials: { accessKeyId: '1', secretAccessKey: '1' },
    region: 'local',
    endpoint: 'http://localhost:8000',
});


/* eslint-disable no-console -- test */
client
    .createTable({
        TableName: 'Test',
        AttributeDefinitions: [{ AttributeName: 'pk', AttributeType: 'S' }],
        KeySchema: [{ AttributeName: 'pk', KeyType: 'HASH' }],
        BillingMode: 'PAY_PER_REQUEST',
    })
    .then(console.log)
    .catch(console.error);
/* eslint-enable no-console -- test */
```

### Observed Behavior

```log
{
  '$metadata': {
    httpStatusCode: 200,
    requestId: '5b2588e6-d92b-4c94-930b-3b5229ab56fb',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  TableDescription: {
    ArchivalSummary: undefined,
    AttributeDefinitions: [ [Object] ],
    BillingModeSummary: {
      BillingMode: 'PAY_PER_REQUEST',
      LastUpdateToPayPerRequestDateTime: 2022-11-13T11:46:59.831Z
    },
    CreationDateTime: 2022-11-13T11:46:59.831Z,
    GlobalSecondaryIndexes: undefined,
    GlobalTableVersion: undefined,
    ItemCount: 0,
    KeySchema: [ [Object] ],
    LatestStreamArn: undefined,
    LatestStreamLabel: undefined,
    LocalSecondaryIndexes: undefined,
    ProvisionedThroughput: {
      LastDecreaseDateTime: 1970-01-01T00:00:00.000Z,
      LastIncreaseDateTime: 1970-01-01T00:00:00.000Z,
      NumberOfDecreasesToday: 0,
      ReadCapacityUnits: 0,
      WriteCapacityUnits: 0
    },
    Replicas: undefined,
    RestoreSummary: undefined,
    SSEDescription: undefined,
    StreamSpecification: undefined,
    TableArn: 'arn:aws:dynamodb:ddblocal:000000000000:table/Test',
    TableClassSummary: undefined,
    TableId: undefined,
    TableName: 'Test',
    TableSizeBytes: 0,
    TableStatus: 'ACTIVE'
  }
}
```

### Expected Behavior

```log
ResourceInUseException: Cannot create preexisting table
    at deserializeAws_json1_0ResourceInUseExceptionResponse (/Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/client-dynamodb/dist-cjs/protocols/Aws_json1_0.js:3027:23)
    at deserializeAws_json1_0CreateTableCommandError (/Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/client-dynamodb/dist-cjs/protocols/Aws_json1_0.js:784:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
    at async /Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:11:20
    at async StandardRetryStrategy.retry (/Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/middleware-retry/dist-cjs/StandardRetryStrategy.js:51:46)
    at async /Users/seb/development/ptrn/paytron/accounts-management/app/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:6:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    requestId: 'd9c22ab3-8673-4284-9e65-23f480d796e7',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  __type: 'com.amazonaws.dynamodb.v20120810#ResourceInUseException'
}
```

### Possible Solution

I don't understand the purpose of overriding the `signingRegion` in [endpoints.js](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-dynamodb/src/endpoint/ruleset.ts#L356), but it seems like either removing that mapping or just replace `us-east-1` with `local` would work well to solve this case.

I assume that there is some reason that this has been done in the first place though. Highlighting this in docs might be good.

**Comment by kuhe** [comment]
I am using the latest AWS CLI v2 (`aws-cli/2.9.5 Python/3.9.11`)
And the latest AWS SDK for JS v3  (`@aws-sdk/client-dynamodb@3.229.0`

In these versions, the "local" region is consistently separate from the us-east-1 region. After creating the table with the CLI in the local region, the JS SDK gets a 400 when attempting to create a table in the same region.

Likewise, querying the table list with the `listTables` operation shows the test table only if the region is set to `local`. 

Test code:
```typescript
import { DynamoDB } from "@aws-sdk/client-dynamodb";
import assert from "assert";

const local = new DynamoDB({
  credentials: { accessKeyId: "1", secretAccessKey: "1" },
  region: "local",
  endpoint: "http://localhost:8000",
});
const east = new DynamoDB({
  credentials: { accessKeyId: "1", secretAccessKey: "1" },
  region: "us-east-1",
  endpoint: "http://localhost:8000",
});

(async () => {
  await local
    .createTable({
      TableName: "TestLocal",
      AttributeDefinitions: [{ AttributeName: "pk", AttributeType: "S" }],
      KeySchema: [{ AttributeName: "pk", KeyType: "HASH" }],
      BillingMode: "PAY_PER_REQUEST",
    })
    .catch(() => {});
  assert((await local.describeTable({ TableName: "TestLocal" })).Table?.TableName === "TestLocal");

  await east
    .createTable({
      TableName: "TestEast",
      AttributeDefinitions: [{ AttributeName: "pk", AttributeType: "S" }],
      KeySchema: [{ AttributeName: "pk", KeyType: "HASH" }],
      BillingMode: "PAY_PER_REQUEST",
    })
    .catch(() => {});
  assert((await east.describeTable({ TableName: "TestEast" })).Table?.TableName === "TestEast");
})();
```

For CLI
```sh
AWS_REGION="local" AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
  aws dynamodb create-table --attribute-definitions '[{"AttributeName":"pk", "AttributeType": "S"}]' \
  --table-name "TestLocal" --key-schema '[{"AttributeName":"pk", "KeyType": "HASH"}]' \
  --endpoint-url "http://localhost:8000" --no-verify-ssl --billing-mode 'PAY_PER_REQUEST'
AWS_REGION="us-east-1" AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
  aws dynamodb create-table --attribute-definitions '[{"AttributeName":"pk", "AttributeType": "S"}]' \
  --table-name "TestEast" --key-schema '[{"AttributeName":"pk", "KeyType": "HASH"}]' \
  --endpoint-url "http://localhost:8000" --no-verify-ssl --billing-mode 'PAY_PER_REQUEST'

AWS_REGION="local" AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
  aws dynamodb list-tables --endpoint-url "http://localhost:8000" \
  --no-verify-ssl
AWS_REGION="us-east-1" AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
  aws dynamodb list-tables --endpoint-url "http://localhost:8000" \
  --no-verify-ssl
```

out:
```
{
    "TableNames": [
        "Test",
        "TestLocal"
    ]
}
{
    "TableNames": [
        "TestEast"
    ]
}
```

I believe the signing region has no impact on the region option used in the clients and operations.

**Comment by sebastian-fredriksson-bernholtz** [comment]
Thanks @kuhe, You are correct, `@aws-sdk/client-dynamodb@3.229.0` is working as expected.

It seems this was fixed between `@aws-sdk/client-dynamodb@3.194.0` and `@aws-sdk/client-dynamodb@3.197.0` (before I opened this issue). I apologise for not confirming that the issue still existed in the latest version before opening this. Closing.

--- 
For reference, I updated your test code to identify the issue just by running the javascript and compared the result between `@aws-sdk/client-dynamodb@3.229.0` and `@aws-sdk/client-dynamodb@3.100.0`. 

The issue is only detectable if `-sharedDb` is false.
```javascript
import { DynamoDB } from "@aws-sdk/client-dynamodb"

const local = new DynamoDB({
	credentials: { accessKeyId: "1", secretAccessKey: "1" },
	region: "local",
	endpoint: "http://localhost:8000",
})
const east = new DynamoDB({
	credentials: { accessKeyId: "1", secretAccessKey: "1" },
	region: "us-east-1",
	endpoint: "http://localhost:8000",
})

;(async () => {
	await local
		.createTable({
			TableName: "Test", // Note that I'm using "Test" table for both clients
			AttributeDefinitions: [{ AttributeName: "pk", AttributeType: "S" }],
			KeySchema: [{ AttributeName: "pk", KeyType: "HASH" }],
			BillingMode: "PAY_PER_REQUEST",
		})
		.then(() => console.log("success local"))
		.catch((e) => console.error("error local"))
	await east
		.createTable({
			TableName: "Test", // Note that I'm using "Test" table for both clients
			AttributeDefinitions: [{ AttributeName: "pk", AttributeType: "S" }],
			KeySchema: [{ AttributeName: "pk", KeyType: "HASH" }],
			BillingMode: "PAY_PER_REQUEST",
		})
		.then(() => console.log("success east"))
		.catch((e) => console.error("error east"))
})()
```

Expected output (which you get with `@aws-sdk/client-dynamodb@3.229.0`):
```
success local
success east
```
Output for `@aws-sdk/client-dynamodb@3.100.0`:
```
success local
error east
```

<details>
<summary>Test run details</Summary>

I ran it twice for each package version just to confirm that there is nothing else that differs. Restarting the container to clear the database in between runs.

```fish
❯ docker run -d -p 8000:8000 amazon/dynamodb-local -jar DynamoDBLocal.jar -inMemory
d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c

❯ npm i @aws-sdk/client-dynamodb@3.100.0

up to date, audited 76 packages in 2s

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ node test-file.mjs
success local
error east

❯ docker restart d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c
d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c

❯ npm i @aws-sdk/client-dynamodb@3.229.0

added 8 packages, removed 5 packages, changed 54 packages, and audited 79 packages in 11s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ node test-file.mjs
success local
success east

❯ docker restart d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c
d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c

❯ npm i @aws-sdk/client-dynamodb@3.100.0
npm WARN deprecated @aws-sdk/util-base64-browser@3.58.0: The package @aws-sdk/util-base64-browser has been renamed to @aws-sdk/util-base64. Please install the renamed package.
npm WARN deprecated @aws-sdk/util-base64-node@3.55.0: The package @aws-sdk/util-base64-node has been renamed to @aws-sdk/util-base64. Please install the renamed package.

added 5 packages, removed 8 packages, changed 54 packages, and audited 76 packages in 8s

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ node test-file.mjs
success local
error east

❯ docker restart d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c
d10bb5e3d782f0fb405395d3830c785be19bb40bab9284249c17ce5fd970af0c

❯ npm i @aws-sdk/client-dynamodb@3.229.0

added 8 packages, removed 5 packages, changed 54 packages, and audited 79 packages in 5s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ node test-file.mjs
success local
success east
```
</details>
